File: java/src/examples/factory/FilePrinter.java
package examples.factory;

import java.io.BufferedWriter;
import java.io.FileWriter;

public class FilePrinter implements Printer {
    private final String fileName;

    public FilePrinter(String destinationName) {
        fileName = destinationName;
    }

    @Override
    public <T> void print(T data) {
        try {
            FileWriter fileWriter = new FileWriter(fileName);
            BufferedWriter out = new BufferedWriter(fileWriter);
            out.write(data.toString());
            out.close();
        } catch (Exception e){
            System.err.println("Couldn't write to file: " + e.getMessage());
        }
    }
}

File: cs/GenericsComparison/Factory/FilePrinter.cs
using System.IO;

namespace Factory
{
    public class FilePrinter : IPrinter
    {
        private readonly string _filename;

        public FilePrinter(string filename)
        {
            this._filename = filename;
        }

        public void Print<T>(T data)
        {
            File.WriteAllText(_filename, data.ToString());
        }
    }
}